cmd/booster-http/gateway_handler.go

```diff
@@ -14,21 +14,23 @@ type gatewayHandler struct {
 	supportedFormats map[string]struct{}
 }
 
-func newGatewayHandler(gw *gateway.BlocksBackend, supportedFormats []string) http.Handler {
+func newGatewayHandler(gw *gateway.BlocksBackend, serveGateway string) http.Handler {
 	headers := map[string][]string{}
+	var deserializedResponse bool
 	gateway.AddAccessControlHeaders(headers)
 
-	fmtsMap := make(map[string]struct{}, len(supportedFormats))
-	for _, f := range supportedFormats {
-		fmtsMap[f] = struct{}{}
+	if serveGateway == "none" {
+		return &gatewayHandler{}
+	} else if serveGateway == "all" {
+		deserializedResponse = true
+	}else if serveGateway == "verifiable" {
+		deserializedResponse = false
 	}
-
 	return &gatewayHandler{
 		gwh: gateway.NewHandler(gateway.Config{
 			Headers:               headers,
-			DeserializedResponses: true,
+			DeserializedResponses: deserializedResponse,
 		}, gw),
-		supportedFormats: fmtsMap,
```

Comment:
If `supportedFormats` is nil, it looks like the code will panic when `h.supportedFormats` is accessed on line 44 (inside `ServeHTTP`)

Reply:
changes this on the latest commit.